### ISSUE TITLE
False positive for F841 in case clause

### ISSUE THREAD

**Issue by Goldziher** [astral-sh/ruff#17477]
### Summary

False positive of unused variable

```python
from typing import Final

EVENT_APPLICATION_SETUP: Final[str] = "application_setup"


def match(event_type: str) -> None:
    match event_type:
        case EVENT_APPLICATION_SETUP: # <-- F841 violation
            ...
```

### Version

>=0.11.1

**Comment by maxmynter** [comment]
I've looked into this a bit and my hypothesis is that while the `visitor` correctly traverses the AST correctly, the semantic analyzer does not add a reference to the name. We seem to only visit the context, not the constant itself. This results in the false positive if the constant is only used in a `match`.

https://github.com/astral-sh/ruff/blob/da6b68cb58824e52efa2a9ae7ebf3c2cbeacad2d/crates/ruff_python_ast/src/visitor.rs#L523-L525
Note that there is no `visitor.visit_expr(value)` call and the `ast::ExprName` also has no`value` field.

I will do some more digging the coming days. But since I'm new to the codebase, I'm thankful for any pointers

**Comment by ntBre** [comment]
I believe this is a duplicate of #16969. Variables in `case` arms must be dotted names if they're meant to refer to existing variables. The `EVENT_APPLICATION_SETUP` in the `case` is a new variable **not** the  `EVENT_APPLICATION_SETUP` outside the `match`, so if you don't use it in the `case` arm it's genuinely unused.

**Comment by maxmynter** [comment]
Seems like it! The following example, using dotted access, doesn't violate F841:

```python
from dataclasses import dataclass
from typing import Final


@dataclass
class Globals:
    EVENT_APPLICATION_SETUP: Final[str] = "application_setup"


def match(event_type: str) -> None:
    match event_type:
        case Globals.EVENT_APPLICATION_SETUP:
            ...
```

Nice, i learned something new today. I didn't use `match` much in Python before.
